arm-storsimple8000series/2017-06-01/swagger/storsimple.json

```diff
@@ -0,0 +1,1778 @@
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/managerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The encryption settings.",
+            "schema": {
+              "$ref": "#/definitions/EncryptionSettings"
+            }
+          }
+        },
+        "deprecated": false
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.StorSimple/managers/{managerName}/extendedInformation/vaultExtendedInfo": {
+      "get": {
+        "tags": [
+          "Managers"
+        ],
+        "description": "Returns the extended information of the specified manager name.",
+        "operationId": "Managers_GetExtendedInfo",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/managerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The manager extended information.",
+            "schema": {
+              "$ref": "#/definitions/ManagerExtendedInfo"
+            }
+          }
+        },
+        "deprecated": false
+      },
+      "put": {
+        "tags": [
+          "Managers"
+        ],
+        "description": "Creates the extended info of the manager.",
+        "operationId": "Managers_CreateExtendedInfo",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "parameters",
+            "in": "body",
+            "description": "The manager extended information.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ManagerExtendedInfo"
+            }
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/managerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The extended information on the manager has been successfully created",
+            "schema": {
+              "$ref": "#/definitions/ManagerExtendedInfo"
```

Comment:
@manaas-microsoft The fact that "x-ms-azure-resource" does flattening of the "properties" property of a Resource is for the convenience of the end user of the SDK, could you clarify your concern about this? Flattening in itself can be accomplished on any model using ["x-ms-client-flatten"](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-client-flatten), which is a separate extension. And I see that you're using it for the "properties" property of ManagerExtendedInfo anyway, so the result will be flattened.
"x-ms-azure-resource" should be used to tag resources (proxy or tracked), this is the guidance we're giving to all services. 
Feel free to generate an SDK with the spec in this PR, using the extension and not using it, and see what version of generated code looks better for you. You can do this by [Installing Autorest](https://github.com/Azure/autorest#installing-autorest) and executing it against your spec, for example to generate C# code "AutoRest.exe -Namespace MyNamespace -Input swagger.json", see [command line reference](https://github.com/Azure/autorest/blob/master/docs/user/cli.md) for more info.